apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -819,78 +820,30 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     end
   end
 
-  def ack_end_device_deletion(realm_name, device_id) do
-    Xandra.Cluster.run(
-      :xandra,
-      &do_ack_end_device_deletion(&1, realm_name, device_id)
-    )
-  end
-
-  defp do_ack_end_device_deletion(conn, realm_name, device_id) do
-    statement = """
-    UPDATE #{realm_name}.deletion_in_progress
-    SET dup_end_ack = true
-    WHERE device_id = :device_id
-    """
-
-    with {:ok, prepared} <- Xandra.prepare(conn, statement),
-         {:ok, %Xandra.Void{}} <-
-           Xandra.execute(conn, prepared, %{"device_id" => device_id}, uuid_format: :binary) do
-      :ok
-    else
-      {:error, %Xandra.Error{} = error} ->
-        _ =
-          Logger.warning(
-            "Database error while writing device deletion end ack: #{Exception.message(error)}"
-          )
-
-        {:error, :database_error}
-
-      {:error, %Xandra.ConnectionError{} = error} ->
-        _ =
-          Logger.warning(
-            "Database connection error while writing device deletion end ack: #{Exception.message(error)}"
-          )
+  def ack_end_device_deletion(keyspace_name, device_id) do
+    query =
+      from(d in DeletionInProgress,
+        prefix: ^keyspace_name,
+        where: d.device_id == ^device_id,
+        update: [set: [dup_end_ack: true]]
+      )
 
-        {:error, :database_connection_error}
-    end
-  end
+    Repo.update_all(query, [], consistency: :each_quorum)
```

Comment:
I think we're not supposed to crash in DUP. Let's use `safe_update_all` instead

```suggestion
    with {:ok, _} <- Repo.update_all(query) do
      :ok
    end
```

make sure to remove the `:ok` at line 833 as well, we should only return `:ok` if the query goes well